src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -3080,8 +3080,18 @@ private void getEEInfo(ref CORINFO_EE_INFO pEEInfoOut)
 
             pEEInfoOut.osPageSize = new UIntPtr(0x1000);
 
-            pEEInfoOut.maxUncheckedOffsetForNullObject = (_compilation.NodeFactory.Target.IsWindows) ?
-                new UIntPtr(32 * 1024 - 1) : new UIntPtr((uint)pEEInfoOut.osPageSize / 2 - 1);
+#if !READYTORUN
+            if (_compilation.TargetArchIsWasm())
+            {
+                // The low 1024 bytes are reserved by Emscripten; no valid address will point there.
+                pEEInfoOut.maxUncheckedOffsetForNullObject = new UIntPtr(1024 - 1);
```

Comment:
It's a good question. I basically took this value from https://github.com/emscripten-core/emscripten/issues/5775#issuecomment-344049528, which suggests that it would be unsupported to compile with Emscripten and use this region.

I would tend towards saying the same for us as well. RyuJit is built to assume this value is "reasonably large", if you set it to something small (like zero), you will encounter correctness issues and code quality will be poor as well.

Reply:
For wasm4 I was not using emscripten for the final link, (this was zero-sharp as well as there is only 64KB to play with).  The global-base required for wasm4 was/is 6560.  So, 1024 would not cause me a problem as any write >= 6560 would be allowed, just the range 1024-6560 would not throw.  We can leave it at 1024 on the basis that it is low enough to allow other runtimes, and the right value for 99% of cases.